File: server/routes/authRoutes.js
const express = require('express');
const router = express.Router();
const authController = require('../controllers/authController');

router.post('/register', authController.register);
router.post('/login', authController.login);
router.post('/forgot-password', authController.forgotPassword);
router.post('/verify-code', authController.verifyCode);       // ✅ Mới: Xác minh OTP với DB
router.post('/reset-password', authController.resetPassword);

module.exports = router;

File: server/routes/auth_routes.py
from fastapi import APIRouter

from controllers import auth_controller
from schemas.auth import (
    ForgotPasswordRequest,
    LoginRequest,
    RegisterRequest,
    ResetPasswordRequest,
)

router = APIRouter()


@router.post("/register")
async def register(payload: RegisterRequest):
    return await auth_controller.register(payload)


@router.post("/login")
async def login(payload: LoginRequest):
    return await auth_controller.login(payload)


@router.post("/forgot-password")
async def forgot_password(payload: ForgotPasswordRequest):
    return await auth_controller.forgot_password(payload)


@router.post("/reset-password")
async def reset_password(payload: ResetPasswordRequest):
    return await auth_controller.reset_password(payload)
